sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/udaf.scala

```diff
@@ -187,7 +187,7 @@ sealed trait BufferSetterGetterUtils {
               row.setNullAt(ordinal)
             }
 
-        case TimestampType =>
+        case TimestampType | TimestampNTZType =>
```

Comment:
This change is unrelated to the bug. However, once I added `TimestampNTZType` to `UnsafeRow.mutableFieldTypes`, the several UDAF-related unit tests in `HashAggregationQuerySuite` pick up this data type and try to test it. However, BufferSetterGetterUtils could not handle this data type (and actually, now that I look, I made the change only for the setter, not the getter).

Alternatively, I could filter out TimestampNTZType in the two HashAggregationQuerySuite unit tests.